docs/integrations/ldap/org.md

```diff
@@ -32,55 +32,30 @@ yarn add @backstage/plugin-catalog-backend-module-ldap
 Update the catalog plugin initialization in your backend to add the provider and
 schedule it:
 
-```ts
-// packages/backend/src/plugins/catalog.ts
-import { CatalogBuilder } from '@backstage/plugin-catalog-backend';
-import { Router } from 'express';
-import { PluginEnvironment } from '../types';
-import { Duration } from 'luxon';
-import { LdapOrgEntityProvider } from '@backstage/plugin-catalog-backend-module-ldap';
-
-export default async function createPlugin(
-  env: PluginEnvironment,
-): Promise<Router> {
-  // The target parameter below needs to match the ldap.providers.target
-  // value specified in your app-config
-  const ldapEntityProvider = LdapOrgEntityProvider.fromConfig(env.config, {
-    id: 'our-ldap-master',
-    target: 'ldaps://ds.example.net',
-    logger: env.logger,
-  });
-
-  const builder = await CatalogBuilder.create(env);
-  builder.addEntityProvider(ldapEntityProvider);
-
-  // You can change the refresh interval for the other catalog entries
-  // independently, or just leave the line below out to use the default
-  // refresh interval. Note that this interval does NOT at all affect
-  // the LDAP refresh when using the provider method, which is good!
-  builder.setRefreshIntervalSeconds(100);
-
-  const { processingEngine, router } = await builder.build();
-  await processingEngine.start();
-
-  // Only perform this scheduling after starting the processing engine
-  await env.scheduler.scheduleTask({
-    id: 'refresh_ldap',
-    // frequency sets how often you want to ingest users and groups from
-    // LDAP, in this case every 60 minutes
-    frequency: Duration.fromObject({ minutes: 60 }),
-    timeout: Duration.fromObject({ minutes: 15 }),
-    fn: async () => {
-      try {
-        await ldapEntityProvider.read();
-      } catch (error) {
-        env.logger.error(error);
-      }
-    },
-  });
-
-  return router;
-}
+```diff
+ // packages/backend/src/plugins/catalog.ts
++import { Duration } from 'luxon';
++import { LdapOrgEntityProvider } from '@backstage/plugin-catalog-backend-module-ldap';
+
+ export default async function createPlugin(
+   env: PluginEnvironment,
+ ): Promise<Router> {
+   const builder = await CatalogBuilder.create(env);
+
++  // The target parameter below needs to match the ldap.providers.target
++  // value specified in your app-config.
++  builder.addEntityProvider(
++    LdapOrgEntityProvider.fromConfig(env.config, {
++      id: 'our-ldap-master',
++      target: 'ldaps://ds.example.net',
++      logger: env.logger,
++      schedule: {
++        scheduler: env.scheduler,
```

Comment:
Initial thing I'm curious of is if there's some kind of pattern where this is inverted a bit so that the provider doesn't depend on the scheduler directly, needing to pass through options etc. I'm thinking along the lines of this:

```ts
{
  ...,
  schedule: env.scheduler.createSchedule({
    frequency: Duration.fromObject({ minutes: 60 }),
    timeout: Duration.fromObject({ minutes: 15 }),
  })
}
```

Reply:
That's basically exactly what I put in the description :) But needs a bit of rewriting of the scheduler first of course. Might make sense to do?